app/scripts/modules/ecs/src/serverGroup/configure/wizard/capacityProvider/CapacityProvider.tsx

```diff
@@ -19,10 +27,42 @@ class EcsCapacityProvider extends React.Component<IEcsCapacityProviderProps, IEc
     const cmd = this.props.command;
 
     this.state = {
-      capacityProviderStrategy: cmd.capacityProviderStrategy,
+      availableCapacityProviders: this.setAvailableCapacityProviders(cmd),
+      defaultCapacityProviderStrategy: this.setDefaultCapacityProviderStrategy(cmd),
+      ecsClusterName: cmd.ecsClusterName,
+      useDefaultCapacityProviders: cmd.useDefaultCapacityProviders || cmd.capacityProviderStrategy && cmd.capacityProviderStrategy.length == 0,
+      capacityProviderStrategy: cmd.capacityProviderStrategy.length > 0 ? cmd.capacityProviderStrategy : [],
+      capacityProviderLoadedFlag: false
     };
   }
 
+  public componentDidMount() {
+    this.props.configureCommand('1').then(() => {
+      const cmd = this.props.command;
+      const useDefaultCapacityProviders = this.state.useDefaultCapacityProviders;
+      this.setState({
+        availableCapacityProviders: this.setAvailableCapacityProviders(cmd),
+        defaultCapacityProviderStrategy: this.setDefaultCapacityProviderStrategy(cmd),
+        capacityProviderStrategy: this.setCapacityProviderStrategy(useDefaultCapacityProviders, cmd),
+        capacityProviderLoadedFlag: true,
+      });
+      this.props.notifyAngular('useDefaultCapacityProviders', this.state.useDefaultCapacityProviders);
+      this.props.notifyAngular('capacityProviderStrategy', this.state.capacityProviderStrategy);
+    });
+  }
+
+  private setAvailableCapacityProviders = (cmd: IEcsServerGroupCommand) => {
+    return cmd.backingData && cmd.backingData.filtered && cmd.backingData.filtered.availableCapacityProviders ? cmd.backingData.filtered.availableCapacityProviders : [];
+  };
+
+  private setDefaultCapacityProviderStrategy = (cmd: IEcsServerGroupCommand) => {
+    return cmd.backingData && cmd.backingData.filtered && cmd.backingData.filtered.defaultCapacityProviderStrategy ? cmd.backingData.filtered.defaultCapacityProviderStrategy : [];
+  };
+
+  private setCapacityProviderStrategy = (useDefaultCapacityProviders: boolean, cmd : IEcsServerGroupCommand) => {
+    return useDefaultCapacityProviders && cmd.backingData && cmd.backingData.filtered && cmd.backingData.filtered.defaultCapacityProviderStrategy ? cmd.backingData.filtered.defaultCapacityProviderStrategy : this.state.capacityProviderStrategy;
+  };
```

Comment:
Could we rename these to get* or retrieve* (or something else). They don't actually set any value, they just return the value.

Reply:
Okay, make sense.